eval.go

```diff
@@ -237,14 +237,28 @@ nextFileMatch:
 		}
 
 		for ; nextDoc < docCount; nextDoc++ {
+			repoID := d.repos[nextDoc]
+			repoMetadata := d.repoMetaData[repoID]
```

Comment:
a bit of a footgun, but you just did a copy of the repository field since the slice is to values not pointers. Previously we didn't do this. To keep the readability and avoid the copy you can do the following
```suggestion
			repoMetadata := &d.repoMetaData[repoID]
```